modules/evolution_engine/orchestrator.py

```
@@ -2,41 +2,68 @@
 from modules.neurons.training.mntp_trainer import MNTPTrainer
 
 logger = logging.getLogger(__name__)
 
 
+@runtime_checkable
+class TrainerProtocol(Protocol):
+    """Protocol describing the trainer expected by the orchestrator."""
+
+    def __init__(
+        self, training_config_path: str, output_dir: str
+    ) -> None:  # noqa: D401
```

Ruff (0.13.1)

Unused `noqa` directive (non-enabled: `D401`)

Remove unused `noqa` directive

(RUF100)